Sources/AsyncAlgorithms/AsyncMapFailureSequence.swift

```diff
@@ -0,0 +1,109 @@
+    /// Converts any failure into a new error.
+    ///
+    /// - Parameter transform: A closure that takes the failure as a parameter and returns a new error.
+    /// - Returns: An asynchronous sequence that maps the error thrown into the one produced by the transform closure.
+    ///
+    /// Use the ``mapFailure(_:)`` operator when you need to replace one error type with another.
+    @available(macOS 15.0, iOS 18.0, watchOS 11.0, tvOS 18.0, visionOS 2.0, *)
+    public func mapFailure<MappedFailure: Error>(_ transform: @Sendable @escaping (Self.Failure) -> MappedFailure) -> some AsyncSequence<Self.Element, MappedFailure> {
+        AsyncMapFailureSequence(base: self, transform: transform)
+    }
+
+    /// Converts any failure into a new error.
+    ///
+    /// - Parameter transform: A closure that takes the failure as a parameter and returns a new error.
+    /// - Returns: An asynchronous sequence that maps the error thrown into the one produced by the transform closure.
+    ///
+    /// Use the ``mapFailure(_:)`` operator when you need to replace one error type with another.
+    @available(macOS 15.0, iOS 18.0, watchOS 11.0, tvOS 18.0, visionOS 2.0, *)
+    public func mapFailure<MappedFailure: Error>(_ transform: @Sendable @escaping (Self.Failure) -> MappedFailure) -> (some AsyncSequence<Self.Element, MappedFailure> & Sendable) where Self: Sendable, Self.Element: Sendable {
+        AsyncMapFailureSequence(base: self, transform: transform)
+    }
+}
+
+/// An asynchronous sequence that converts any failure into a new error.
+@available(macOS 15.0, iOS 18.0, watchOS 11.0, tvOS 18.0, visionOS 2.0, *)
+fileprivate struct AsyncMapFailureSequence<Base: AsyncSequence, MappedFailure: Error>: AsyncSequence {
+
+    typealias AsyncIterator = Iterator
+    typealias Element = Base.Element
+    typealias Failure = Base.Failure
+
+    private let base: Base
+    private let transform: @Sendable (Failure) -> MappedFailure
+
+    init(
+        base: Base,
+        transform: @Sendable @escaping (Failure) -> MappedFailure
+    ) {
+        self.base = base
+        self.transform = transform
+    }
+
+    func makeAsyncIterator() -> Iterator {
+        Iterator(
+            base: base.makeAsyncIterator(),
+            transform: transform
+        )
+    }
+}
+
+@available(macOS 15.0, iOS 18.0, watchOS 11.0, tvOS 18.0, visionOS 2.0, *)
+extension AsyncMapFailureSequence {
+
+    /// The iterator that produces elements of the map sequence.
+    fileprivate struct Iterator: AsyncIteratorProtocol {
+
+        typealias Element = Base.Element
+
+        private var base: Base.AsyncIterator
+
+        private let transform: @Sendable (Failure) -> MappedFailure
+
+        init(
+            base: Base.AsyncIterator,
+            transform: @Sendable @escaping (Failure) -> MappedFailure
+        ) {
+            self.base = base
+            self.transform = transform
+        }
+
+        mutating func next() async throws(MappedFailure) -> Element? {
```

Comment:
very minor nit: you could just have this `next` function call the `next(isolation: nil)` to save on potential refactoring points.

not a blocker imho

Reply:
oh yes! updated!